internal/fs/monitoring_fs.go

```diff
@@ -0,0 +1,342 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fs
+
+import (
+	"context"
+	"time"
+
+	"github.com/jacobsa/fuse/fuseops"
+	"github.com/jacobsa/fuse/fuseutil"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	counterFsRequests = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "gcsfuse_fs_requests",
+			Help: "Number of requests per file system API.",
+		},
+		[]string{ // labels
+			"method",
+		},
+	)
+	counterFsErrors = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "gcsfuse_fs_errors",
+			Help: "Number of errors per file system API.",
+		},
+		[]string{ // labels
+			"method",
+		},
+	)
+	latencyOpenFile = prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Name:    "gcsfuse_fs_open_file_latency",
+			Help:    "The latency of executing an OpenFile request in ms.",
+			Buckets: prometheus.ExponentialBuckets(0.01, 10, 8),
+		},
+	)
+	latencyReadFile = prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Name:    "gcsfuse_fs_read_file_latency",
+			Help:    "The latency of executing a ReadFile request in ms.",
+			Buckets: prometheus.ExponentialBuckets(0.01, 10, 8),
```

Comment:
Please add a comment to explain what buckets are generated. I usually use:
// [0.1, 1.5, 30] generates 30 buckets between 0.1 ms and 12 seconds.